src/videojs-hls.js

```diff
@@ -1030,6 +1030,11 @@ videojs.Hls.translateMediaIndex = function(mediaIndex, original, update) {
     // let the media index be zero when there are no segments defined
     return 0;
   }
+  
+  if (original.mediaSequence === update.mediaSequence) {
+	// playlist sequence didn't change
+	return mediaIndex;
```

Comment:
Ok, that makes sense. It seems to me that the conditional is wrong. If we made the test `translatedMediaIndex >= update.segments.length + 1`, would that resolve the issue?